examples/pipelines/commoncrawl/components/download_commoncrawl_segments/src/main.py

```diff
@@ -0,0 +1,81 @@
+import logging
+import typing as t
+
+import io
+import boto3
+
+import dask.bag as db
+import dask.dataframe as dd
+from dask.delayed import delayed
+
+import pandas as pd
+
+import requests
+from warcio.archiveiterator import ArchiveIterator
+
+from fondant.component import DaskTransformComponent
+
+logger = logging.getLogger(__name__)
+
+BASE_URL = "https://data.commoncrawl.org/"
+
+
+def get_records_from_warc_file(
+    warc_file: str, n_records_to_download: t.Optional[int] = None
+) -> t.List:
+    """Downloads a WARC file and extracts the webpages.
+    Args:
+        warc_file: The path to the WARC file.
+        n_records_to_download: The number of webpages to download from the WARC file.
+    Returns:
+        A list of webpages.
+    """
+    logger.info(f"Processing WARC file from segment path: {warc_file}...")
+    records = []
+    counter = 0
+    response = requests.get(BASE_URL + warc_file, stream=True)
+    response.raise_for_status()
+
+    for record in ArchiveIterator(response.raw, arc2warc=True):
+        if record.rec_type == "response":
+            url = record.rec_headers.get_header("WARC-Target-URI")
+            content = (
```

Comment:
Do you think it would be possible to add an optional possibility to return plain text instead of the html? 
We could add an additional parameter if html or plain text should be returned. I think you don't need it for your use case, but for the large language model use case it would be super useful! :)

For the plain text transformation you could use BeautifulSoup. Some time ago I did something similar by using this code function: 
```
    def _convert_to_plain_text(html):
        """Convert html body into plain text. Making sure table rows are on seperate lines."""
        soup = BeautifulSoup(html)
        body_content = soup.find("body")
        text = ""

        for e in body_content.descendants:
            if isinstance(e, str):
                text += e.strip()
            elif e.name in ['br', 'h1', 'h2', 'h3', 'h4', 'tr', 'th']:
                text += '\n'
            elif e.name in ['td']:
                text += ' '
            elif e.name == 'p' and not any(parent.name == 'table' for parent in e.parents):
                text += '\n'
            elif e.name in ['td']:
                text += '\t'
            elif e.name == 'li':
                text += '\n- '

        return text
```

Reply:
Is `soup.get_text()` not sufficient?

Reply:
Probably. The custom code handles table a bit differently. Basically, try to keep the table structure line by line.

Reply:
I see the custom code keeps the CSS content in the <style> tag. I'm assuming you don't want this. Will `soup.get_text()` be ok for you @mrchtr ? otherwise, we'll have a long list of tags to parse

Reply:
`soup.get_text()` works! Even better as my custom code.